### DISCUSSION TITLE
Set custom Server header for all pages

### DISCUSSION THREAD

**Discussion by ririko5834** [NginxProxyManager/nginx-proxy-manager#1926] [Q&A]
![image](https://user-images.githubusercontent.com/64729293/158020555-74d95d26-5cea-46a8-bba7-970c743bda4d.png)
How can I set custom Server header for all pages. If server header is not possible to change, how can I add custom header that will be on all proxy hosts, redirects, streams, 404 hosts, etc.?

**Comment by mochman** [comment]
Have you tried adding a custom config as [described here](https://nginxproxymanager.com/advanced-config/#custom-nginx-configurations)?

Edit:
Looks like #1061 is an Issue already.  There seems to be a solution that requires building your own openresty.

**Reply by ririko5834** [reply]
When I want to add just header powered-by to every page, what do I need to do?

**Reply by mochman** [reply]
Like [the issue states](https://github.com/NginxProxyManager/nginx-proxy-manager/issues/1061#issuecomment-834149210), you need to clone the [docker-nginx-full](https://github.com/NginxProxyManager/docker-nginx-full) add the line `--build=powered-by` to the configuration section of [openresty](https://github.com/NginxProxyManager/docker-nginx-full/blob/master/scripts/build-openresty).  Then rebuild all the docker images.

**Reply by ririko5834** [reply]
I dont need to change Server header, but just to add custom header to all proxy hosts and redirects - https://nginxproxymanager.com/advanced-config/#custom-nginx-configurations
I don't understand this. Can you tell me where to add my custom config and how it should look?

**Reply by mochman** [reply]
Create a file `/data/nginx/custom/server_proxy.conf` and `/data/nginx/custom/server_redirect.conf` in your container and put in your headers.
Something like:
```
add_header YOUR_HEADER
```
or:
```
proxy_set_header YOUR_HEADER
```
depending on if you want the header going to the client or local backend server.
